src/Compilation.zig

```diff
@@ -5870,6 +5963,17 @@ pub fn addCCArgs(
                         // Without this flag, Clang would invoke UBSAN when such an extern
                         // function was called.
                         try argv.append("-fno-sanitize=function");
+
+                        // It's recommended to use the minimal runtime in production environments
+                        // due to the security implications of the full runtime. The minimal runtime
+                        // doesn't provide much benefit over simply trapping.
+                        if (mod.optimize_mode == .ReleaseSafe) {
+                            try argv.append("-fsanitize-trap=undefined");
+                        }
+
+                        if (target.os.tag == .windows) {
```

Comment:
```suggestion
                        // This is necessary because, by default, Clang instructs LLVM to embed a COFF link dependency on
                        // `libclang_rt.ubsan_standalone.a` when the UBSan runtime is used.
                        if (target.os.tag == .windows) {
```